exec/cnex/lib/file_posix.c

```diff
@@ -0,0 +1,383 @@
+#ifdef __APPLE__
+#include <copyfile.h>
+#else
+// The following is needed for fchmod() on Posix
+#define  _POSIX_C_SOURCE 200809L
+#endif
+#include <dirent.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <utime.h>
+
+#include "enums.h"
+#include "exec.h"
+#include "cell.h"
+#include "nstring.h"
+#include "stack.h"
+
+
+
+static void handle_error(TExecutor *exec, int error, const char *path)
+{
+    switch (error) {
+        case EACCES: exec->rtl_raise(exec, "FileException.PermissionDenied", path, BID_ZERO);      break;
+        case EEXIST: exec->rtl_raise(exec, "FileException.DirectoryExists", path, BID_ZERO);       break;
+        case ENOENT: exec->rtl_raise(exec, "FileException.PathNotFound", path, BID_ZERO);          break;
+        default: {
+            char err[1024];
+            snprintf(err, 1024, "%s: %s", path, strerror(error));
```

Comment:
Isn't there a constant size you can use here instead of hard-coding a specific numeric value? Also, use `sizeof(err)` in the `snprintf` call so you don't have to repeat the size.

Reply:
Used PATH_MAX for the size of err.